File: app/src/main/java/com/bopr/android/smailer/ui/OptionsFragment.java
package com.bopr.android.smailer.ui;

import android.os.Bundle;

import androidx.preference.EditTextPreference;
import androidx.preference.ListPreference;
import androidx.preference.Preference;

import com.bopr.android.smailer.R;
import com.bopr.android.smailer.util.AndroidUtil;

import static com.bopr.android.smailer.Settings.KEY_PREF_DEVICE_ALIAS;
import static com.bopr.android.smailer.Settings.KEY_PREF_EMAIL_LOCALE;
import static com.bopr.android.smailer.util.Util.isEmpty;

/**
 * Options settings activity's fragment.
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
@Deprecated
public class OptionsFragment extends BasePreferenceFragment {

    private BaseSettingsListener settingsListener;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        settingsListener = new BaseSettingsListener(requireContext());
        settings.registerOnSharedPreferenceChangeListener(settingsListener);
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.pref_options);

        findPreference(KEY_PREF_EMAIL_LOCALE).setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
            @Override
            public boolean onPreferenceChange(Preference preference, Object value) {
                updateLocalePreference((ListPreference) preference, (String) value);
                return true;
            }
        });

        findPreference(KEY_PREF_DEVICE_ALIAS).setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {

            @Override
            public boolean onPreferenceChange(Preference preference, Object value) {
                updateAlasPreference((EditTextPreference) preference, (String) value);
                return true;
            }
        });

    }

    @Override
    public void onDestroy() {
        settings.unregisterOnSharedPreferenceChangeListener(settingsListener);
        super.onDestroy();
    }

    private void updateLocalePreference(ListPreference preference, String value) {
        int index = preference.findIndexOfValue(value);
        if (index < 0) {
            updateSummary(preference, getString(R.string.not_specified), STYLE_ACCENTED);
        } else {
            CharSequence cs = preference.getEntries()[index];
            updateSummary(preference, cs.toString(), STYLE_DEFAULT);
        }
    }

    private void updateAlasPreference(EditTextPreference preference, String value) {
        if (isEmpty(value)) {
            updateSummary(preference, AndroidUtil.getDeviceName(), STYLE_DEFAULT);
        } else {
            updateSummary(preference, value, STYLE_DEFAULT);
        }
    }

}

File: app/src/main/kotlin/com/bopr/android/smailer/ui/OptionsFragment.kt
package com.bopr.android.smailer.ui

import android.os.Bundle
import androidx.preference.Preference
import com.bopr.android.smailer.CallProcessor
import com.bopr.android.smailer.PhoneEvent
import com.bopr.android.smailer.R
import com.bopr.android.smailer.Settings
import com.bopr.android.smailer.ui.BatteryOptimizationHelper.BATTERY_OPTIMIZATION_DIALOG_TAG
import com.bopr.android.smailer.util.deviceName
import com.bopr.android.smailer.util.runLongTask
import com.bopr.android.smailer.util.showToast
import java.lang.System.currentTimeMillis

/**
 * Options fragment.
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
class OptionsFragment : BasePreferenceFragment() {

    override fun onCreatePreferences(bundle: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.pref_options)

        requirePreference("reset_dialogs").onPreferenceClickListener =
            Preference.OnPreferenceClickListener {
                onResetDialogs()
                true
            }

        requirePreference("sent_test_email").onPreferenceClickListener =
            Preference.OnPreferenceClickListener {
                onSendTestEmail(it)
                true
            }
    }

    private fun onResetDialogs() {
        Settings(requireContext()).update {
            remove(BATTERY_OPTIMIZATION_DIALOG_TAG)
        }
        showToast(R.string.operation_complete)
    }

    private fun onSendTestEmail(preference: Preference) {
        preference.runLongTask(
            onPerform = {
                CallProcessor(requireContext()).process(
                    PhoneEvent(
                        phone = getString(R.string.app_name),
                        isIncoming = true,
                        startTime = currentTimeMillis(),
                        endTime = currentTimeMillis(),
                        text = "Sample message",
                        acceptor = deviceName()
                    )
                )
            },
            onComplete = { _, _ ->
                /* NOTE: if we live the page while processing context becomes null */
                context?.showToast(R.string.operation_complete)
            })
    }

}